src/java/org/apache/cassandra/io/sstable/SSTableRewriter.java

```diff
@@ -224,18 +225,7 @@ private void moveStarts(SSTableReader newReader, DecoratedKey lowerbound)
             return;
 
         newReader.setupOnline();
-        List<DecoratedKey> invalidateKeys = null;
```

Comment:
why do we no longer need to invalidate keys?

Reply:
It seems I forgot a `break` in the migration code, and I'm rerunning the benchmark, also with a better configuration for iterator compaction so that migration does not put it over the edge.

If the results show an improvement in latency or a low enough overhead, I will bring back migration before committing.

Reply:
The migration tests are [ready](https://fallout.sjc.dsinternal.org/performance/branimir.lambov@datastax.com/report/45aa5681-4831-44b1-a936-12bbb577fc08).

I do not see any improvement in [latency](https://fallout.sjc.dsinternal.org/artifacts/performance_reports/branimir.lambov@datastax.com/45aa5681-4831-44b1-a936-12bbb577fc08/report.html?metric=Latency_99th&operation=latency_1_1%3Alatency_1_1.result-success&operation_filter=.%2Bresult-success&smoothing=1&show_aggregates=true&min_ops=0&xmin=0&xmax=5159.3&ymin=0&ymax=40) with cache migration enabled, while there is a very visible deterioration in [throughput](https://fallout.sjc.dsinternal.org/artifacts/performance_reports/branimir.lambov@datastax.com/45aa5681-4831-44b1-a936-12bbb577fc08/report.html?metric=Ops_Sec&operation=throughput%3Athroughput.result-success&operation_filter=.%2Bresult-success&smoothing=1&show_aggregates=true&min_ops=0&xmin=0&xmax=61149.63&ymin=0&ymax=364312.3) (250k -> 220k for cursors, 197k -> 180k for iterators) caused by significantly slowed down compaction.

To improve compaction performance and simplify configuration, I am going to remove the feature. Please let me know if you do not agree with this decision.